src/lib.rs

```diff
@@ -134,19 +141,15 @@ impl Datadog {
         let _ = INSTANCE.get_or_try_init::<_, Error>(|| {
             initialized = true;
 
-            if configuration.is_reporting_enabled() {
-                let dogstatsd_client_options: dogstatsd::Options = dogstatsd::Options::new(
-                    configuration.from_addr(),
-                    configuration.to_addr(),
-                    configuration.namespace(),
-                    configuration.default_tags(),
-                );
-
-                let client: dogstatsd::Client = dogstatsd::Client::new(dogstatsd_client_options)?;
-                Ok(Self::new(client))
-            } else {
-                Ok(Self { client_opt: None })
-            }
+            let dogstatsd_client_options: dogstatsd::Options = dogstatsd::Options::new(
+                configuration.from_addr(),
+                configuration.to_addr(),
+                configuration.namespace(),
+                configuration.default_tags(),
+            );
+
+            let client: dogstatsd::Client = dogstatsd::Client::new(dogstatsd_client_options)?;
+            Ok(Self::new(client, configuration.is_reporting_enabled()))
```

Comment:
The two things should live together because we (prima.it) have a custom configuration that avoid reporting if the environment is `Qa` ([here](https://github.com/primait/prima_datadog.rs/blob/master/src/configuration/prima.rs#L39)). But maybe we can find a workaroud for this?

Reply:
I see, it makes sense then 😄